iroh/src/client.rs

```diff
@@ -104,37 +178,99 @@ where
         };
         Ok(Some(doc))
     }
+}
+
+/// Iroh blobs client.
+#[derive(Debug, Clone)]
+pub struct BlobsClient<C> {
+    rpc: RpcClient<ProviderService, C>,
+}
+
+impl<C> BlobsClient<C>
+where
+    C: ServiceConnection<ProviderService>,
+{
+    /// Read a single blob as a streaming [`BlobReader`].
+    pub async fn read(&self, hash: Hash) -> Result<BlobReader> {
+        BlobReader::from_rpc(&self.rpc, hash).await
+    }
 
-    /// Get the bytes for a hash.
+    /// Read all bytes of single blob.
     ///
-    /// Note: This reads the full blob into memory.
-    pub async fn get_bytes(&self, hash: Hash) -> Result<Bytes> {
-        let mut stream = self.get_bytes_stream(hash).await?;
-        stream.read_to_end().await
+    /// This allocates a buffer for the full blob. Use only if you know that the blob you're
+    /// reading is small. If not sure, use [`Self::read`] and check the size with
+    /// [`BlobReader::size`] before calling [`BlobReader::read_to_end`].
+    pub async fn read_to_end(&self, hash: Hash) -> Result<Bytes> {
```

Comment:
Maybe `read_to_bytes` matching the `read_to_vec` convention I have seen used before?